megatron/ds_text_generation.py

```diff
@@ -0,0 +1,317 @@
+# Copyright (c) 2023, NVIDIA CORPORATION. All rights reserved.
+
+"""Pretrain utilities."""
+
+from datetime import datetime
+import math
+import sys
+import time
+import json
+from functools import partial
+# The earliest we can measure the start time.
+_TRAIN_START_TIME = time.time()
+import torch
+from torch.nn.parallel.distributed import DistributedDataParallel as torchDDP
+
+from megatron import get_args
```

Comment:
Merge the imports as

from megatron import (
    get_args,
    get_times,
    ....
)